ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGen/policyBuilder.go

```diff
@@ -37,7 +36,7 @@ func (pbuilder *PolicyBuilder) Build(customResourseOnly bool) (map[string]interf
 			if err != nil {
 				panic(err)
 			}
-			_, resourceDef := pbuilder.getCustomResource(sFile.Data, sFile.Spec, sPolicyFile, rname, pbuilder.PolicyGenTemp.Metadata.Labels.Mcp)
+			_, resourceDef := pbuilder.getCustomResource(sFile.Data, sFile.Spec, sFile.Labels,sPolicyFile, rname, pbuilder.PolicyGenTemp.Metadata.Labels.Mcp)
```

Comment:
well, that will make getCustomResource() depend on the sourceFile struct and changing the struct may require changes inside the method not just as passing param. I guess its better to make it more generic for the input param ?

Reply:
I prefer to pass in the struct as it makes it easy in the future to make additional data available without having to update the method signature (set of arguments). If the getCustomResource method is called from multiple places with inputs coming from very different types of data then it may make sense to separate out the fields like this, but given how you use it I would suggest passing in the struct. I'm approving the PR but it is something to consider in further refinements.